src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -127,18 +127,56 @@ public static bool TryCreateNativeIntegerType(INamedTypeSymbol symbol, out TypeS
 
 #if !CODE_STYLE
 
+#nullable enable
             public override TypeSyntax VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
             {
-                // TODO(https://github.com/dotnet/roslyn/issues/39865): generate the calling convention once exposed through the API
+                FunctionPointerCallingConventionSyntax? callingConventionSyntax = null;
+                if (symbol.Signature.CallingConvention != System.Reflection.Metadata.SignatureCallingConvention.Default)
+                {
+                    SeparatedSyntaxList<FunctionPointerUnmanagedCallingConventionSyntax> conventionsList = default;
+                    switch (symbol.Signature.CallingConvention)
+                    {
+                        case System.Reflection.Metadata.SignatureCallingConvention.CDecl:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Cdecl") });
+                            break;
+                        case System.Reflection.Metadata.SignatureCallingConvention.StdCall:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Stdcall") });
+                            break;
+                        case System.Reflection.Metadata.SignatureCallingConvention.ThisCall:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Thiscall") });
+                            break;
+                        case System.Reflection.Metadata.SignatureCallingConvention.FastCall:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Fastcall") });
+                            break;
+
+                        default:
+                            // All types that come from CallingConventionTypes start with "CallConv". We don't want the prefix for the actual
+                            // syntax, so strip it off
+                            const int CallConvLength = 8;
+                            conventionsList = SyntaxFactory.SeparatedList(symbol.Signature.CallingConventionTypes.SelectAsArray(type => GetConventionForString(type.Name[CallConvLength..])));
+                            break;
+                    }
+
+                    callingConventionSyntax = SyntaxFactory.FunctionPointerCallingConvention(
+                        SyntaxFactory.Token(SyntaxKind.UnmanagedKeyword),
+                        conventionsList.Count > 0
+                            ? SyntaxFactory.FunctionPointerUnmanagedCallingConventionList(conventionsList)
+                            : null);
+
+                    static FunctionPointerUnmanagedCallingConventionSyntax GetConventionForString(string identifier)
+                        => SyntaxFactory.FunctionPointerUnmanagedCallingConvention(SyntaxFactory.Identifier(identifier));
```

Comment:
Should SyntaxFactory just have an overload for FunctionPointerUnmanagedCallingConvention that does this?

Reply:
Maybe. I'm not super convinced we need to maintain an explicitly-added method for this though.

---
In reply to: [463857164](https://github.com/dotnet/roslyn/pull/46467#discussion_r463857164) [](ancestors = 463857164)